compiler/codegen/cg-symbol.cpp

```diff
@@ -144,6 +145,15 @@ bool shouldLlvmPrintIrFn(FnSymbol* fn) {
   return shouldLlvmPrintIrName(fn->name) || shouldLlvmPrintIrCName(fn->cname);
 }
 
+std::vector<const char*> gatherPrintLlvmIrCNames() {
+  std::vector<const char*> ret;
+  for (auto elt : llvmPrintIrCNames) {
+    ret.push_back(elt);
+  }
+
+  return ret;
+}
+
```

Comment:
What's the motivation for exposing this function returning a vector instead of just exposing the `llvmPrintIrCNames` set directly?

Reply:
Do you mean, why not return a `std::set` instead of a `std::vector`, or why have a function to return it at all, instead of just exposing the global variable?

Why not the global variable? Seems like nicer style to have functions.

Why return a vector instead of a set? The code calling it will just process them in order and so the set nature is not particularly important. But it would be fine to have it return a set if that becomes important for some reason.

Reply:
Yeah, either why not just expose the global or have this return the set if you don't want to expose the global. 

> Why not the global variable? Seems like nicer style to have functions.

I don't disagree with that, I was just curious since the callsite used the global `llvmPrintIrStageNum` but made a wrapper for this so it wasn't obvious to me why there was a difference.

> Why return a vector instead of a set? The code calling it will just process them in order and so the set nature is not particularly important. But it would be fine to have it return a set if that becomes important for some reason.

But the order in the vector is just the iteration order of the set, right? I guess I don't see the point in copying the set into a vector instead of just directly using the set (this is minor and I'm fine leaving it as-is too, I just wasn't sure if I was missing something.)